### PR TITLE
Make sure scan can handle failed transaction gracefully

### LINKED ISSUE #1034: Make sure scan can handle failed transaction gracefully
After #1010 is in, failed transactions' messages won't be complete and will not have some fields. Make sure scan can handle that.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1034 

please test in:
RPC_URL=http://54.169.193.107/rest/ GRAPHQL_URL=ws://54.169.193.107/v1/graphql yarn parcel index.html --no-cache

### CODE REVIEW

File: chain/db/db.go [hunk 1]
```diff
@@ -283,7 +290,9 @@ func (b *BandDB) HandleTransactionFail(tx auth.StdTx, txHash []byte) {
 		message["type"] = txMsg.Type()
 		messages = append(messages, message)
 	}
-	b.UpdateTransaction(txHash, messages)
+
+	wrapedMsg := wrapMessage(messages, "fail")
```
Comment (sorawit) [inline chain/db/db.go]: failure

File: scan/src/subscriptions/TxSub.re [hunk 2]
```diff
@@ -282,7 +296,17 @@ module Msg = {
...
   };
-  let decodeActions = json => json |> JsonUtils.Decode.list(decodeAction);
+
+  let decodeFailAction = json => FailMessage(json |> FailMessage.decode);
+
+  let decodeActions = json =>
+    JsonUtils.Decode.(
+      switch (json |> field("status", string)) {
+      | "success" => json |> field("messages", list(decodeAction))
+      | "fail" => json |> field("messages", list(decodeFailAction))
```
Comment (sorawit) [inline scan/src/subscriptions/TxSub.re]: ditto here.

File: scan/src/subscriptions/TxSub.re [hunk 3]
```diff
@@ -282,7 +296,17 @@ module Msg = {
...
+      | _ => []
```
Comment (sorawit) [inline scan/src/subscriptions/TxSub.re]: Let's do `exception` so we fail fast.

**Review by sorawit — DISMISSED** [review]

**Review by taobun — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
We can apply `makeBadge()` in a success case. However, we can do it on the next ticket.

**Pull request merged**